internal/ls/lsutil/userpreferences.go

```diff
@@ -38,87 +48,85 @@ type UserPreferences struct {
 	// Enables auto-import-style completions on partially-typed import statements. E.g., allows
 	// `import write|` to be completed to `import { writeFile } from "fs"`.
-	IncludeCompletionsForImportStatements core.Tristate
+	IncludeCompletionsForImportStatements core.Tristate `pref:"suggest.includeCompletionsForImportStatements"`
 	// Unless this option is `false`,  member completion lists triggered with `.` will include entries
 	// on potentially-null and potentially-undefined values, with insertion text to replace
 	// preceding `.` tokens with `?.`.
-	IncludeAutomaticOptionalChainCompletions core.Tristate
+	IncludeAutomaticOptionalChainCompletions core.Tristate `pref:"suggest.includeAutomaticOptionalChainCompletions"`
 	// Allows completions to be formatted with snippet text, indicated by `CompletionItem["isSnippet"]`.
 	IncludeCompletionsWithSnippetText core.Tristate // !!!
 	// If enabled, completions for class members (e.g. methods and properties) will include
 	// a whole declaration for the member.
 	// E.g., `class A { f| }` could be completed to `class A { foo(): number {} }`, instead of
 	// `class A { foo }`.
-	IncludeCompletionsWithClassMemberSnippets core.Tristate // !!!
+	IncludeCompletionsWithClassMemberSnippets core.Tristate `pref:"suggest.classMemberSnippets.enabled"` // !!!
 	// If enabled, object literal methods will have a method declaration completion entry in addition
 	// to the regular completion entry containing just the method name.
 	// E.g., `const objectLiteral: T = { f| }` could be completed to `const objectLiteral: T = { foo(): void {} }`,
 	// in addition to `const objectLiteral: T = { foo }`.
-	IncludeCompletionsWithObjectLiteralMethodSnippets core.Tristate // !!!
-	JsxAttributeCompletionStyle                       JsxAttributeCompletionStyle
+	IncludeCompletionsWithObjectLiteralMethodSnippets core.Tristate               `pref:"suggest.objectLiteralMethodSnippets.enabled"` // !!!
+	JsxAttributeCompletionStyle                       JsxAttributeCompletionStyle `pref:"preferences.jsxAttributeCompletionStyle"`
 
 	// ------- AutoImports --------
 
-	ImportModuleSpecifierPreference modulespecifiers.ImportModuleSpecifierPreference // !!!
-	// Determines whether we import `foo/index.ts` as "foo", "foo/index", or "foo/index.js"
-	ImportModuleSpecifierEnding       modulespecifiers.ImportModuleSpecifierEndingPreference // !!!
-	IncludePackageJsonAutoImports     IncludePackageJsonAutoImports                          // !!!
-	AutoImportSpecifierExcludeRegexes []string                                               // !!!
-	AutoImportFileExcludePatterns     []string                                               // !!!
-	PreferTypeOnlyAutoImports         bool                                                   // !!!
+	ModuleSpecifier ModuleSpecifierUserPreferences
+
+	IncludePackageJsonAutoImports IncludePackageJsonAutoImports `pref:"preferences.includePackageJsonAutoImports"` // !!!
```

Comment:
Do you think it would be better to just omit these until the code is used?

Reply:
IIRC this is upcoming, so it would just be churn to remove I think...